src/modules/assets/assets.entity.ts

```diff
@@ -0,0 +1,101 @@
+import { Field, ID, ObjectType } from '@nestjs/graphql';
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  OneToMany,
+  OneToOne,
+  PrimaryColumn,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+import { DIDDocumentEntity } from '../did/did.entity';
+import { AssetHistoryEvent, AssetHistoryEventType } from './assets.event';
+
+@ObjectType()
+@Entity()
+export class Asset {
+  static create(data: Partial<Asset>) {
+    const entity = new Asset();
+    Object.assign(entity, data);
+    return entity;
+  }
+  @PrimaryColumn({ type: 'text' })
+  @Field(() => ID)
+  id: string;
+
+  @Column()
+  @Field()
+  owner: string;
+
+  @Column({ nullable: true })
+  @Field({ nullable: true })
+  offeredTo?: string;
+
+  @Column()
+  @Field()
+  createdAt: string;
+
+  @Column()
+  @Field()
+  updatedAt: string;
+
+  @OneToOne(
+    () => DIDDocumentEntity,
+    did => did.id,
+    { eager: true },
+  )
+  @JoinColumn()
+  @Field(() => DIDDocumentEntity)
+  document: DIDDocumentEntity;
+
+  @OneToMany(
+    () => AssetsHistory,
+    history => history.asset,
+    { eager: true },
+  )
+  history: AssetsHistory[];
+}
+
+@ObjectType()
+@Entity()
+export class AssetsHistory implements AssetHistoryEvent {
+  static create(data: Omit<AssetsHistory, 'id'>) {
+    const entity = new AssetsHistory();
+    Object.assign(entity, data);
+    return entity;
+  }
+  @PrimaryGeneratedColumn()
+  @Field(() => ID)
+  id: number;
+
+  @Column()
+  @Field()
+  emittedBy: string;
+
+  @Column({ nullable: true })
+  @Field({ nullable: true })
+  relatedTo?: string;
+
+  @Column({ type: 'enum', enum: AssetHistoryEventType })
+  @Field()
+  type: AssetHistoryEventType;
+
+  @Column()
+  @Field()
+  at: number;
+
+  @Column()
+  @Field()
+  timestamp: string;
+
+  @ManyToOne(
+    () => Asset,
+    asset => asset.history,
+  )
+  asset?: Asset;
```

Comment:
In what cases would `AssetHistory` not have an associated asset?

Reply:
Normally we won't have the `assetId` field in the entity, cause it is created by the typeorm to keep the foreign key. But this has a couple of downsides, to create a relation you need to pass the whole asset entity, even if you already have the assetId and you don't want to perform a request to db to get it. Second downside is that you're not able to filter the entity by the assetId field, cause TypeOrm doesnt not this field. So to overcome this I've added a special field named `assetId` and now we're able to create relationship with asset just be passing the `assetId` and we're able to filter by this value :)

Reply:
I've also added a typechecker to check if one of this fields passed to entity creator:
```ts
Omit<AssetsHistory, 'id'> & ({ asset: Asset } | { assetId: string })
```

Reply:
Ok, thanks